### PR TITLE
Fix Media Element crash in IOS with Sample app

### LINKED ISSUE #1562: [BUG] MediaElement Crash when entering Full Screen
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [X] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

When clicking on full screen button in IOS the app pauses and AVPLayer controls stop responding. This does not affect the app just the controls. The only control that still works is exit full screen button. You have to exit page and return to have working media element. It does go full screen but playback stops.

### Expected Behavior

When clicking on full screen it should go full screen and continue playback.

### Steps To Reproduce

1: launch Community toolkit sample. 
2: Navigate to media element page.
3: Click on full screen button.

### Link to public reproduction project repository

https://github.com/CommunityToolkit/Maui/tree/main/samples

### Environment

```markdown
- .NET MAUI CommunityToolkit: Current Main, not nuget.
- OS: IOS 17.x
- .NET MAUI: 8.0.x
```

### Anything else?

Not sure what is causing this. It has been tested against current up to date IOS on a physical device with current main build from November 23rd 2023. Dotnet 8.0.x. Using CommunityToolkit Sample app. I have been trying to figure out the cause but I am currently using windows and cannot debug it at all. There is no error message it just shows paused as current status. The only thing I noticed that was odd was position tracking broke when switching to full screen. If I test against #1555 I keep position tracking though it still crashes. I am very puzzled by this behavior.

### PR DESCRIPTION
**Pull request by ne0rrmatrix**
- Bug fix

 ### Description of Change ###

Fix Media Element Crash in IOS when going fullscreen or exiting full screen. This issue affect IOS 17.x. Outside the sample app the issue does not appear to be reproducible. 

 ### Linked Issues ###
 - Fixes #1562 

 ### PR Checklist ###

 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [ ] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pulls 

 ### Additional information ###
 Not sure if we want to override OnDisappearing or some else like OnNavigatingFrom? Input would be greatly appreciated. If unloaded event is used it is triggered by going full screen.

### CODE REVIEW

**Comment by ne0rrmatrix** [comment]
Do we want to make changes in the documentation and change unloaded event to OnDissapearing?

**Comment by jfversluis** [comment]
Would you be able to test 16 easily?

File: samples/CommunityToolkit.Maui.Sample/Pages/Views/MediaElement/MediaElementPage.xaml.cs [hunk 1]
```diff
@@ -114,12 +114,11 @@ void OnMuteClicked(object? sender, EventArgs e)
 		MediaElement.ShouldMute = !MediaElement.ShouldMute;
 	}
 
-	void BasePage_Unloaded(object? sender, EventArgs e)
+	protected override void OnDisappearing()
```
Comment (pictos) [inline samples/CommunityToolkit.Maui.Sample/Pages/Views/MediaElement/MediaElementPage.xaml.cs:117]: Please, do not use `OnAppearing/Disappearing` use the `OnNavigatedTo/From` instead (:

**Comment by ne0rrmatrix** [comment]
> Would you be able to test 16 easily?

I can test on a simulator. But my Ipad is on IOS 17.x. I would not want to trust my mac mini to behave properly. It is old and runs very slowly. 2014 model year. So not even supported but still running latest. I only use it for building.

**Comment by ne0rrmatrix** [comment]
Should we be running our PRs against iOS 16.x and 17.x for testing purposes? I have an iPad running iOS 17 ATM.

**Comment by pictos** [comment]
> For future reference. Should we be running our PRs against iOS 16.x and 17.x for testing purposes? I have an iPad running iOS 17 ATM.

the ideal would be to test on all supported versions, but we don't have time/budget for that... So testing on just one it's fine, two is amazing

**Comment by ne0rrmatrix** [comment]
Tested now on IOS 16.4 and 17.x. The fix works for both. Testing against main both IOS version have the issue where when clicking on full screen it unloads and disposes media element. The above fix resolves the issue on both versions.

**Review by jfversluis — APPROVED** [review]

**Pull request merged**